seabird.proto

```diff
@@ -4,168 +4,196 @@ option go_package = ".;pb";
 
 package seabird;
 
-// Identity is included in every message that requires auth. One AuthMethod at a
-// time is supported. For now, the only supported method is a token. These need
-// to be created manually and distributed to any clients.
-message Identity {
-  oneof AuthMethod {
-    string token = 1;
-  }
-}
+import "common.proto";
 
-service Seabird {
-  rpc StreamEvents(StreamEventsRequest) returns (stream Event);
+// Base objects are split into two pieces - their identifier and info
+// components. The info objects embed the identifier to make them easier to send
+// in responses.
 
-  // Chat actions
-  rpc SendMessage(SendMessageRequest) returns (SendMessageResponse);
-  rpc JoinChannel(JoinChannelRequest) returns (JoinChannelResponse);
-  rpc LeaveChannel(LeaveChannelRequest) returns (LeaveChannelResponse);
+message Service {
+  string id = 1;
+}
 
-  // Chat connection introspection
-  rpc ListChannels(ListChannelsRequest) returns (ListChannelsResponse);
-  rpc GetChannelInfo(ChannelInfoRequest) returns (ChannelInfoResponse);
-  rpc SetChannelTopic(SetChannelTopicRequest) returns (SetChannelTopicResponse);
+message ServiceInfo {
+  Service service = 1;
+  string type = 2;
+}
 
-  // Seabird introspection
-  rpc ListStreams(ListStreamsRequest) returns (ListStreamsResponse);
-  rpc GetStreamInfo(StreamInfoRequest) returns (StreamInfoResponse);
-  rpc GetCoreInfo(CoreInfoRequest) returns (CoreInfoResponse);
+message User {
+  string service_id = 1;
+  string user_id = 2;
 }
 
-message StreamEventsRequest {
-  Identity identity = 1;
+message UserInfo {
+  User user = 1;
+  string name = 2;
+}
 
-  map<string, CommandMetadata> commands = 2;
+message Channel {
+  string service_id = 1;
+  string channel_id = 2;
 }
 
-message CommandMetadata {
-  string name = 1;
-  string short_help = 2;
-  string full_help = 3;
+message ChannelInfo {
+  Channel channel = 1;
+  string name = 2;
+  string topic = 3;
 }
```

Comment:
Each of these names sound a bit backwards to me. A name like `User` suggests that it's a structure containing all relevant information about a user, and is basically what you're calling `UserInfo`. Instead, I'd do this:

```protobuf
message UserSelector {
  string service_id = 1;
  string user_id = 2;
}

message User {
  UserSelector selector = 1;
  string name = 2;
}
```

This makes it clearer (to me) when to expect or pass selection information like IDs versus expecting or passing full structures.

Reply:
I like that - I'll make those changes for all the relevant structs.

Reply:
These were all changed and cleaned up.